scripts/wallet.js

```diff
@@ -252,6 +266,38 @@ export class Wallet {
     async getKeyToExport() {
         return await this.#masterKey?.getKeyToExport(this.#nAccount);
     }
+
+    //TODO: for future return only the enum item and not the address path
+    async isMyVout(script) {
+        let addresses = [];
+        const dataBytes = hexToBytes(script);
+        if (isP2PKH(dataBytes)) {
+            addresses.push(this.updatePkhMap(bytesToHex(dataBytes.slice(3, 23))))
+            const path = await this.isOwnAddress(addresses[0]);
+            if (path) {
+                return [UTXO_WALLET_STATE.SPENDABLE, path];
+            }
+        } else if (isP2CS(dataBytes)) {
+            addresses.push(this.updatePkhMap(bytesToHex((dataBytes.slice(6, 26)))));
+            addresses.push(this.updatePkhMap(bytesToHex((dataBytes.slice(28, 48)))))
+            const path1 = await this.isOwnAddress(addresses[0]);
+            const path2 = await this.isOwnAddress(addresses[1]);
```

Comment:
These should be clarified, `path1` and `path2` is not very clear without reading the script templates.